Add tunable to allow squid listen on snmp port

TomasKorbar opened the pull request:
Squid is able to provide statistics through snmp protocol so there should be option for its selinux policy to allow this.

This change allows squid to listen and reply on standard snmp ports.

Resolves #1601

TomasKorbar committed:
Add tunable to allow squid bind snmp port

Squid is able to provide statistics through snmp protocol
so there should be option for its selinux policy to allow this.

This change allows squid to bind standard snmp ports.

time->Tue Feb 14 07:55:40 2023
type=PROCTITLE msg=audit(1676379340.080:654): proctitle=2873717569642D636F6F72642D3929002D2D6B69640073717569642D636F6F72642D39002D2D666F726567726F756E64002D66002F6574632F73717569642F73717569642E636F6E66
type=SOCKADDR msg=audit(1676379340.080:654): saddr=0A0000A1000000000000000000000000000000000000000000000000
type=SYSCALL msg=audit(1676379340.080:654): arch=c000003e syscall=49 success=no exit=-13 a0=d a1=560012f71810 a2=1c a3=7ffe13546f54 items=0 ppid=8287 pid=8289 auid=4294967295 uid=23 gid=23 euid=0 suid=0 fsuid=0 egid=23 sgid=23 fsgid=23 tty=(none) ses=4294967295 comm="squid" exe="/usr/sbin/squid" subj=system_u:system_r:squid_t:s0 key=(null)
type=AVC msg=audit(1676379340.080:654): avc:  denied  { name_bind } for  pid=8289 comm="squid" src=161 scontext=system_u:system_r:squid_t:s0 tcontext=system_u:object_r:snmp_port_t:s0 tclass=udp_socket permissive=0

Resolves #1601

zpytela commented:
@TomasKorbar The description says listen, but the PR content has also bind, so all permissions are needed? Can you also add at least one of the (interpreted) denials into the commit message?

TomasKorbar commented:
@zpytela there were indeed unnecessary permissions included. I removed them. It seems that snmp is working with just bind permission. Also i added the AVC log to commit message.

zpytela commented:
Thank you; the commit header and description should also be adjusted accordingly. And how about naming the boolean differently, e. g. squid_bind_snmp_port? See
```
# semanage boolean -l | grep _bind_
```

TomasKorbar commented:
@zpytela Both done.

zpytela commented:
Thank you for your effort, merging.
